lib/transformers/DockerCompose.js

```diff
@@ -286,8 +286,34 @@ DockerComposeTransformer.prototype.validateIntrusiveFeatures = function(yamlStri
     }
 
     try {
-        var compositionObject = YAML.load(yamlString);
+        var compositionObject = YAML.safeLoad(yamlString);
+
+        // remove all services that were marked by the engine explicitly to not be validated
+        if (_.get(compositionObject, 'version', '').toString() === '2') {
+            compositionObject.services = _.filter(compositionObject.services, function(service) {
+                return !service.dontValidateIntrusiveFeatures;
+            });
+        } else {
+            compositionObject = _.filter(compositionObject, function(service) {
+                return !service.dontValidateIntrusiveFeatures;
+            });
+        }
+
         IntrusiveFeaturesValidator.validate(compositionObject);
+
+        // remove all occurencees of dontValidateIntrusiveFeatures field on services
+        compositionObject = YAML.safeLoad(yamlString);
```

Comment:
Why do we need to reparse the yaml string after validation? we already parsed it